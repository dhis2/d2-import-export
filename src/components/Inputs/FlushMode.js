import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_AUTO = { value: 'AUTO', label: i18n.t('Auto') }
export const OPTION_OBJECT = { value: 'OBJECT', label: i18n.t('Object') }
export const FLUSH_MODE_KEY = 'flushMode'
export const FLUSH_MODE_DEFAULT_VALUE = OPTION_AUTO.value

export const FlushMode = () => (
    <RadioGroup
        name={FLUSH_MODE_KEY}
        label={i18n.t('Flush mode')}
        options={[OPTION_AUTO, OPTION_OBJECT]}
    />
)
