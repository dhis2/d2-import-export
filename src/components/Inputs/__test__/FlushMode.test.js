import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { FlushMode } from '../FlushMode'

jest.mock('react-final-form')

describe('Input component - FlushMode', () => {
    it('should render correctly', () => {
        const file = shallow(<FlushMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
