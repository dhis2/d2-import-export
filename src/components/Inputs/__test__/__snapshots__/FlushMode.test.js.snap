// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - FlushMode should render correctly 1`] = `
<RadioGroup
  label="Flush mode"
  name="flushMode"
  options={
    Array [
      Object {
        "label": "Auto",
        "value": "AUTO",
      },
      Object {
        "label": "Object",
        "value": "OBJECT",
      },
    ]
  }
/>
`;
